### PR TITLE
bump version numbers in 3.1 release branch

### PR DESCRIPTION
**Pull request by khieta**
## Description of changes

Update Cargo.toml version numbers and CHANGELOGs on the `release/3.1.x` branch. We need to remember to pull this edit to the `main` branch once the new version is released.

## Disclaimer

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Review by aaronjeline — APPROVED** [review]

**Review by john-h-kastner-aws — APPROVED** [review]

**Comment by khieta** [comment]
Note: downstream dependency failures are expected since I haven't created `release/3.1.x` branches in those repositories yet. Will work on this today.

File: cedar-policy-core/Cargo.toml [hunk 1]
```diff
@@ -13,7 +13,7 @@ repository = "https://github.com/cedar-policy/cedar"
 
 [dependencies]
 serde = { version = "1.0", features = ["derive", "rc"] }
-serde_with = { version = "3.0", features = ["json"] }
+serde_with = { version = "3.1", features = ["json"] }
```
Comment (cdisselkoen) [inline cedar-policy-core/Cargo.toml:16]: intentional?

File: cedar-policy/CHANGELOG.md [hunk 2]
```diff
@@ -5,7 +5,8 @@ All notable changes to this project will be documented in this file.
 The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
 and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).
 
-## [Unreleased]
+## [3.1.0] - coming soon
+Cedar Language Version: 3.0.0
```
Comment (cdisselkoen) [inline cedar-policy/CHANGELOG.md:9]: I think this should be 3.1.0.  New schema syntax is a language minor

File: cedar-policy/CHANGELOG.md [hunk 3]
```diff
@@ -370,7 +371,7 @@ Cedar Language Version: 2.0.0
 Cedar Language Version: 2.0.0
 - Initial release of `cedar-policy`.
 
-[Unreleased]: https://github.com/cedar-policy/cedar/compare/v3.0.1...main
+[3.1.0]: https://github.com/cedar-policy/cedar/compare/v3.0.1...v3.1.0
```
Comment (cdisselkoen) [inline cedar-policy/CHANGELOG.md:374]: Let's keep `[Unreleased]` as well, and just have it start from 3.1.0

**Review by cdisselkoen — CHANGES_REQUESTED** [review]
Doing request-changes just for the comments in cedar-policy/CHANGELOG.md

File: cedar-policy-core/Cargo.toml (hunk 1 above)
Comment (khieta) [reply cedar-policy-core/Cargo.toml:16]: Nope!

File: cedar-policy/CHANGELOG.md (hunk 2 above)
Comment (khieta) [reply cedar-policy/CHANGELOG.md:9]: Ack. Makes sense.

**Review by cdisselkoen — APPROVED** [review]

**Pull request merged**